pkg/controller/backup_finalizer_controller.go

```diff
@@ -243,6 +244,12 @@ func (r *backupFinalizerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		if len(volumeSnapshots) > 0 && len(volumeSnapshotContents) > 0 {
 			deleteVolumeSnapshots(volumeSnapshots, volumeSnapshotContents, log, 30, r.client, r.volumeSnapshotClient, 10)
 		}
+
+		// check if the VSBs associated with the backup are in completed state, if yes then clean them up
+		err = datamover.DeleteVSBsIfComplete(backup.Name)
```

Comment:
would it be better to pass r.client instead of making a new one?

Reply:
not sure if you could here since VSBs are not in the same ns

Reply:
For clientsets maybe.. this is normal kbclient
There is nothing ns specific about this. You still have to specify namespace in the object.metadata during get/create/update
```
func GetVolumeSnapMoverClient() (kbclient.Client, error) {
	client, err := kbclient.New(config.GetConfigOrDie(), kbclient.Options{})
	if err != nil {
		return nil, err
	}
	err = snapmoverv1alpha1.AddToScheme(client.Scheme())
	if err != nil {
		return nil, err
	}

	return client, err
}
```